deepspeed/runtime/zero/stage3.py

```diff
@@ -2109,6 +2148,26 @@ def get_fp32_grad_partitions(self) -> Dict[int, Dict[int, Tensor]]:
 
         return grad_dict
 
+    def get_fp32_grad_for_param(self, param) -> Tensor:
+        self.__reduce_and_partition_stream.synchronize()
+        reduce_buffer = torch.zeros(param.ds_shape, dtype=torch.float32, device=param.device).flatten()
+
+        if self.offload_optimizer:
+            group_idx, dest_offset, num_elements = self.grad_position[self.get_param_id(param)]
+            fp32_grad = self.fp32_partitioned_groups_flat[group_idx].grad.narrow(
+                0,
+                dest_offset,
+                num_elements)            
```

Comment:
```suggestion
            fp32_grad = self.fp32_partitioned_groups_flat[group_idx].grad.narrow(
                0,
                dest_offset,
                num_elements) .to(device=param.device)
```

This is needed for the offload to work